src/Datadog.Trace/AppSec/Waf/NativeBindings/Native.cs

```diff
@@ -39,43 +57,200 @@ internal class Native
+                    path,
+                    Path.Combine(path, runtimeId),
+                    Path.Combine(path, runtimeId, "native"),
+                };
+            }
+
+            // treat any path that could contain integrations.json as home folder
+            var integrationsPaths = Environment.GetEnvironmentVariable("DD_INTEGRATIONS")
+                    ?.Split(';')
+                    ?.Where(x => !string.IsNullOrWhiteSpace(x))
+                    ?.Select(Path.GetDirectoryName)
+                    ?.ToList()
+                        ?? new List<string>();
+
+            // the real trace home
+            var tracerHome = Environment.GetEnvironmentVariable("DD_DOTNET_TRACER_HOME");
+            if (!string.IsNullOrWhiteSpace(tracerHome))
+            {
+                integrationsPaths.Add(tracerHome);
+            }
+
+            // include the appdomain base as this will help framework samples running in IIS find the library
+            var currentDir =
+                string.IsNullOrEmpty(AppDomain.CurrentDomain.RelativeSearchPath) ?
+                    AppDomain.CurrentDomain.BaseDirectory : AppDomain.CurrentDomain.RelativeSearchPath;
+            integrationsPaths.Add(currentDir);
+
+            // the home folder could contain the native dll directly, but it could also
+            // be under a runtime specific folder
+            var paths =
+                integrationsPaths
+                    .Distinct()
+                    .SelectMany(GetRuntimeSpecificVersions)
+                    .ToList();
+
+            return paths;
+        }
+
+        private static void SearchPathsAndLoadLibrary(string libName, List<string> paths)
+        {
+            var success = false;
+            foreach (var path in paths)
+            {
+                var libFullPath = Path.Combine(path, libName);
+
+                if (!File.Exists(libFullPath))
+                {
+                    continue;
+                }
+
+                // loading the library is sufficient, once in memory the p/invokes will just work
+                var loaded = NativeLibrary.TryLoad(libFullPath, out var _);
+
+                if (loaded)
+                {
+                    success = true;
+                    Log.Information($"Loaded library '{libName}' from '{path}'");
+                    break;
+                }
+                else
+                {
+                    Log.Warning($"Failed to load library '{libName}' from '{path}'");
+                }
+            }
+
+            if (!success)
+            {
+                Log.Warning($"Failed to load library '{libName}' from any of the following '{string.Join(", ", paths)}'");
+            }
+        }
+
+        private static void GetLibNameAndRuntimeId(out string libName, out string runtimeId)
+        {
+            string runtimeIdPart1, libPrefix, libExt;
+
+            switch (Environment.OSVersion.Platform)
+            {
+                case PlatformID.MacOSX:
```

Comment:
[The documentation](https://docs.microsoft.com/en-us/dotnet/api/system.platformid) says this value was only returned in Silverlight. .NET Core on macOS returns `Unix`. You want want to use you `FrameworkDescription` helper instead.
https://github.com/DataDog/dd-trace-dotnet/blob/beba073dc8f9dd86b5f9becdb19feb4bdb8b69a0/src/Datadog.Trace/FrameworkDescription.cs#L16

Reply:
fixed.